File: sample/src/main/java/com/stepstone/stepper/sample/step/fragment/StepperFeedbackStepFragment.java
/*
Copyright 2017 StepStone Services

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
 */

package com.stepstone.stepper.sample.step.fragment;

import android.os.Bundle;
import android.os.Handler;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.annotation.UiThread;
import android.view.View;
import android.widget.TextView;

import com.stepstone.stepper.BlockingStep;
import com.stepstone.stepper.StepperLayout;
import com.stepstone.stepper.VerificationError;
import com.stepstone.stepper.sample.R;

import butterknife.Bind;

public class StepperFeedbackStepFragment extends ButterKnifeFragment implements BlockingStep {

    private static final String STEP_POSITION = "step_position";

    public static StepperFeedbackStepFragment newInstance(int stepPosition) {
        StepperFeedbackStepFragment fragment = new StepperFeedbackStepFragment();
        Bundle arguments = new Bundle();
        arguments.putInt(STEP_POSITION, stepPosition);
        fragment.setArguments(arguments);
        return fragment;
    }

    @Bind(R.id.stepContent)
    TextView stepContent;

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        stepContent.setText("Step content #" + getArguments().getInt(STEP_POSITION) + "\n" + getString(R.string.lorem_ipsum));
    }

    @Override
    public VerificationError verifyStep() {
        return null;
    }

    @Override
    public void onSelected() {
    }

    @Override
    public void onError(@NonNull VerificationError error) {
    }

    @Override
    @UiThread
    public void onNextClicked(final StepperLayout.OnNextClickedCallback callback) {
        callback.getStepperLayout().showProgress("Operation in progress, please wait...");
        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                callback.goToNextStep();
                callback.getStepperLayout().hideProgress();
            }
        }, 2000L);
    }

    @Override
    public void onCompleteClicked(StepperLayout.OnCompleteClickedCallback callback) {
       callback.complete();
    }

    @Override
    @UiThread
    public void onBackClicked(StepperLayout.OnBackClickedCallback callback) {
        callback.goToPrevStep();
    }

    @Override
    protected int getLayoutResId() {
        return R.layout.fragment_step_stepper_feedback;
    }
}

File: sample/src/main/java/com/stepstone/stepper/sample/step/fragment/StepperFeedbackStepFragment.kt
/*
Copyright 2017 StepStone Services

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
 */

package com.stepstone.stepper.sample.step.fragment

import android.annotation.SuppressLint
import android.os.Bundle
import android.os.Handler
import android.support.annotation.UiThread
import android.view.View
import android.widget.TextView

import com.stepstone.stepper.BlockingStep
import com.stepstone.stepper.StepperLayout
import com.stepstone.stepper.VerificationError
import com.stepstone.stepper.sample.R

import butterknife.BindView

internal class StepperFeedbackStepFragment : ButterKnifeFragment(), BlockingStep {

    companion object {

        private const val STEP_POSITION = "step_position"

        fun newInstance(stepPosition: Int): StepperFeedbackStepFragment {
            val fragment = StepperFeedbackStepFragment()
            val arguments = Bundle()
            arguments.putInt(STEP_POSITION, stepPosition)
            fragment.arguments = arguments
            return fragment
        }
    }

    @BindView(R.id.stepContent)
    lateinit var stepContent: TextView

    @SuppressLint("SetTextI18n")
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        stepContent.text = "Step content #${arguments!!.getInt(STEP_POSITION)} \n ${getString(R.string.lorem_ipsum)}"
    }

    override fun verifyStep(): VerificationError? {
        return null
    }

    override fun onSelected() {}

    override fun onError(error: VerificationError) {}

    @UiThread
    override fun onNextClicked(callback: StepperLayout.OnNextClickedCallback) {
        callback.stepperLayout.showProgress("Operation in progress, please wait...")
        Handler().postDelayed({
            callback.goToNextStep()
            callback.stepperLayout.hideProgress()
        }, 2000L)
    }

    override fun onCompleteClicked(callback: StepperLayout.OnCompleteClickedCallback) {
        callback.complete()
    }

    @UiThread
    override fun onBackClicked(callback: StepperLayout.OnBackClickedCallback) {
        callback.goToPrevStep()
    }

    override val layoutResId: Int
        get() = R.layout.fragment_with_text_content

}
